Classes/Private/Libxml2/Data/CommentNode.swift

```diff
@@ -0,0 +1,93 @@
+import Foundation
+
+extension Libxml2 {
+    /// Comment nodes use to hold HTML comments like this: <!-- This is a comment -->
+    ///
+    class CommentNode: Node {
+
+        var text: String
+
+        init(text: String) {
+            self.text = text
+
+            super.init(name: "text")
+        }
+
+        override func customMirror() -> Mirror {
+            return Mirror(self, children: ["type": "text", "name": name, "text": text, "parent": parent.debugDescription], ancestorRepresentation: .Suppressed)
+        }
+
+        /// Node length.
+        ///
+        override func length() -> Int {
+            return text.characters.count
```

Comment:
The length should be zero for this node, since it is intended for content between the opening and closing tags.  In fact if you run the demo app, and delete a range of text after the comment in this PR you'll notice than when switching to HTML the removed range will be a different one (due to misalignment).

This is a more graphical explanation of what length represents:

```html
<b>...child nodes go here...</b>: length is the sum of the lengths of all child nodes
<textNode>Hello there!</textNode>: length is the length of "Hello there!"
<br>: length is zero since there's no opening and closing tag
<!-- Some comment -->: length is zero since there's no opening and closing tag
```

We can rename `length()` to something more specific if necessary.